gcc/rust/typecheck/rust-tyty.h

```diff
@@ -135,6 +135,11 @@ class TypeBoundsMappings
   std::vector<TypeBoundPredicate> specified_bounds;
 };
 
+extern void
+set_cmp_autoderef_mode ();
```

Comment:
This is the hack i mentioned @CohenArthur I would be much easier for us to live with this hack for now. Its going to take a few PR's to get our method resolution closer to that match statement in Rustc